### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by elia**
## Description

This is an extraction from https://github.com/solidusio/solidus_starter_frontend/pull/278 of things not strictly related to supporting solidus_paypal_commerce_platform and are of general utility.

## How Has This Been Tested?

CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: bin/rake
Comment (gsmendoza) [inline bin/rake:1]: Duplicate "standard" in "Add standard standard rake tasks and bin/setup" commit message.

File: template.rb
Comment (gsmendoza) [inline template.rb:75]: Can you elaborate on this comment? It mentions several things (system tests gem, minitest, `skip-test`, "both scenarios"), but I don't understand how they relate with capybara and the other test gems below.

**Review by gsmendoza — CHANGES_REQUESTED** [review]

File: Rakefile
Comment (waiting-for-dev) [inline Rakefile:1]: For context, how did you generate these files?

File: bin/sandbox
Comment (waiting-for-dev) [inline bin/sandbox:27]: Could we add more information to the message of this commit? I think we're doing several things here:

- Installing after solidus
- Cleaning unneeded gems
- Acknowledging that solidus_starter_frontend's main branch only works for solidus master branch

Comment (waiting-for-dev) [reply bin/sandbox:27]: Ideally, we should split it in different commits, but a better explanation can be enough 🙂

File: templates/spec/support/solidus_starter_frontend/capybara.rb
Comment (waiting-for-dev) [inline templates/spec/support/solidus_starter_frontend/capybara.rb:15]: I'm ok removing apparition, but I'm afraid we're also losing docker support for tests here. Did you check if that's still working?

**Review by waiting-for-dev — CHANGES_REQUESTED** [review]
Thanks, @elia. I'd left some considerations.

File: template.rb
Comment (elia) [reply template.rb:75]: I'm replacing it with:

```
    # We need to add capybara along with a javascript driver to support the provided system specs,
    # `rails new` will add the following gems for system tests unless `--skip-test` is provided.
    # We want to stick with them but we can't be sure about how the app was generated, so we'll
    # add them only if they're not already in the Gemfile.
```

hope it makes more sense, but basically you'll get those three gems in your pristine `rails new` gemfile depending on which flags you provided. We also want them as they're the default for rails testing but we need to take into account that they might be already there.

File: Rakefile
Comment (elia) [reply Rakefile:1]: I created it manually, the `with_unbundled_env` part by trial and error.

In case you were interested specifically in `rake/clean`, this is standard Rake, and it provides the `clobber` task borrowing the term from `make`, see [Clobbering - Wikipedia](https://en.wikipedia.org/wiki/Clobbering)
https://en.wikipedia.org/wiki/Clobbering

File: template.rb
Comment (gsmendoza) [inline template.rb:73]:
```suggestion
    # We need to add capybara along with a javascript driver to support the provided system specs.
```

Period instead of comma?

**Review by gsmendoza — CHANGES_REQUESTED** [review]

File: template.rb
Comment (elia) [reply template.rb]: Period is fine 👍 (pushed)

File: templates/spec/support/solidus_starter_frontend/capybara.rb
Comment (waiting-for-dev) [reply templates/spec/support/solidus_starter_frontend/capybara.rb:15]: Yeah, I think it's better to fix Docker on another PR. But let's rebase this one on top of the other (or wait until it's merged and rebase from main). I want to be sure that we're not breaking tests on Docker here, as I'm quite confident that removing the custom options will break it.

I've created the PR to fix Docker to speed up things and not block you. Please, could you check that https://github.com/solidusio/solidus_starter_frontend/pull/280 works for you? 🙏 I've cherry-picked your commit installing default-mysql-client, and I've also added libvips to the image. However, I can't reproduce your issue with nokogiri.

File: bin/sandbox
Comment (waiting-for-dev) [reply bin/sandbox:27]: Thanks!

File: template.rb
Comment (waiting-for-dev) [inline template.rb:79]: I'd love to get rid of this hack, as that's something that end users will see. On top of that, I've experienced `Bundler.locked_gems` to be very slow sometimes.

What about modifying our spec helper with something like this for each case?

```ruby
begin
  require 'capybara'
rescue LoadError
  raise "SSF needs capybara to be available for.... Please add it to the the `:test` group in your Gemfile"
end
```

Comment (waiting-for-dev) [inline template.rb:79]: I'd love to get rid of this hack, as that's something that end users will see. On top of that, I've experienced `Bundler.locked_gems` to be very slow sometimes.

What about modifying our spec helper with something like this for each case?

```ruby
begin
  require 'capybara'
rescue LoadError
  raise "SSF needs capybara to be available for.... Please add it to the the `:test` group in your Gemfile"
end
```

**Review by waiting-for-dev — CHANGES_REQUESTED** [review]

File: templates/app/views/products/_product_info.html.erb
Comment (waiting-for-dev) [inline templates/app/views/products/_product_info.html.erb:3]: Awesome! I saw there were some warnings because of that while running tests ❤️

File: template.rb
Comment (elia) [reply template.rb:79]: > I'd love to get rid of this hack, as that's something that end users will see.

This won't be seen by the user, as it's happening in the app-template, that `gem` DSL is not the same as the one we have in a Gemfile. The slowness probably happens when the lock is not there already and needs to be generated (just guessing).

The proposed solution could work but given we're not dirtying anything in the user Gemfile I would consider a better experience for everything to be working out of the box (at least for the most beaten paths).

Comment (waiting-for-dev) [reply template.rb:79]: Oh, yeah, sorry, I got confused. Yeah, if that doesn't pollute the user's Gemfile, that's the best solution 👍 Sorry for the noise.

**Review by elia — COMMENTED** [review]
@waiting-for-dev you think we can merge this and possibly fix the docker development env in a separate PR?

**Comment by waiting-for-dev** [comment]
Hey @elia, I'm ok as long as we keep the custom options for the driver, although I'm ok getting rid of apparition. We need to fix Docker on ARM, but if we merge the as it is now we'll be introducing a regression for x86. At a later stage, we can also try to remove those custom options if we find a way. Does it make sense?

File: templates/spec/support/solidus_starter_frontend/capybara.rb
Comment (elia) [reply templates/spec/support/solidus_starter_frontend/capybara.rb:23]: Sorry, I missed one line while copy/pasting from capybara, now should be fixed

Comment (elia) [reply templates/spec/support/solidus_starter_frontend/capybara.rb:23]: It seems to be working fine, I also added bin/rspec so it's easier to run failed specs from the rspec output

**Review by waiting-for-dev — APPROVED** [review]
We finally got rid of the apparition driver but kept the custom option so that Docker is not broken on x86. Fixing Docker for ARM is still pending.

**Pull request merged**
